src/components/TransactionsTable.vue

```diff
@@ -132,17 +136,56 @@ export default defineComponent({
               );
       }
       if (tableData) {
-        rows.value = tableData.map(
-          (tx) =>
-            ({
-              name: tx.trx_id,
-              transaction: { id: tx.trx_id, type: 'transaction' },
-              timestamp: tx['@timestamp'],
-              action: tx,
+        const map = new Map();
+        tableData.forEach((item) => {
+          const key = item.trx_id;
+          const collection = map.get(key) as {
+            timestamp: string;
+            name: string;
+            actions: TransactionTableActionRow[];
+          };
+          if (!collection) {
+            map.set(key, {
+              name: item.trx_id,
+              transaction: { id: item.trx_id, type: 'transaction' },
+              timestamp: item['@timestamp'],
+              action: item,
+              data: hasActions.value
+                ? { data: item.data as unknown, name: item.account as unknown }
+                : { data: item.act.data as unknown, name: item.act.name },
+              actions: [
+                {
+                  name: item.trx_id,
+                  transaction: { id: item.trx_id, type: 'transaction' },
+                  timestamp: item['@timestamp'],
+                  action: item,
+                  data: hasActions.value
+                    ? {
+                        data: item.data as unknown,
+                        name: item.account as unknown
+                      }
+                    : { data: item.act.data as unknown, name: item.act.name }
+                }
+              ]
+            });
+          } else {
+            collection.actions.push({
+              name: item.trx_id,
+              transaction: { id: item.trx_id, type: 'transaction' },
+              timestamp: item['@timestamp'],
+              action: item,
               data: hasActions.value
-                ? { data: tx.data, name: tx.account }
-                : { data: tx.act.data, name: tx.act.name }
-            } as TransactionTableRow)
+                ? {
+                    data: item.act.data as unknown,
+                    name: item.account
+                  }
+                : { data: item.act.data as unknown, name: item.act.name }
+            });
+          }
+        });
+        rows.value = Array.from(
+          map,
+          ([, value]) => value as TransactionTableRow
         );
       }
       void filterRows();
```

Comment:
@WikusKriek can you add  test coverage for `paginationSettings()` and `filterRows()`? refactor as necessary.